src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2999,23 +2999,53 @@ private void VisitStatementsWithLocalFunctions(BoundBlock block)
             if (!TrackingRegions && !block.LocalFunctions.IsDefaultOrEmpty)
```

Comment:
>!TrackingRegions [](http://example.com/codeflow?start=16&length=16)

It looks like this condition is always true, let's convert it into an assert. This will make it clear that we will never get into the `else` block with a local function among the statements.